ml_metadata/tools/mlmd_bench/fill_nodes_workload.cc

```diff
@@ -0,0 +1,321 @@
+      (*node_batch[i].mutable_properties())[it->first].set_string_value(
+          property_value);
+      curr_num_properties++;
+      it++;
+    }
+    // If the node's number of properties is greater than the type(the iterator
+    // of the type properties has reached the end, but `curr_num_properties`
+    // is still less than `num_properties`), uses custom properties instead.
+    while (curr_num_properties < num_properties) {
+      (*node_batch[i]
+            .mutable_custom_properties())[absl::StrCat("custom_p-",
+                                                       curr_num_properties)]
+          .set_string_value(property_value);
+      curr_num_properties++;
+    }
+    curr_bytes += GetTransferredBytes<NT>(node_batch[i]);
+  }
+}
+
+// Sets additional fields(uri, state) for artifacts and returns transferred
+// bytes for these additional fields.
+int64 SetArtifactsAdditionalFields(
+    const std::string& node_batch_name,
+    google::protobuf::RepeatedPtrField<Artifact>& node_batch) {
+  int64 bytes = 0;
+  for (int64 i = 0; i < node_batch.size(); ++i) {
+    node_batch[i].set_uri(absl::StrCat(node_batch_name, "_node_", i, "_uri"));
+    node_batch[i].set_state(Artifact::UNKNOWN);
+    bytes += node_batch[i].uri().size() + 1;
+  }
+  return bytes;
+}
+
+// Sets additional field(state) for executions and returns transferred
+// bytes for these additional fields.
+int64 SetExecutionsAdditionalFields(
+    google::protobuf::RepeatedPtrField<Execution>& node_batch) {
+  int64 bytes = 0;
+  for (int64 i = 0; i < node_batch.size(); ++i) {
+    node_batch[i].set_last_known_state(Execution::UNKNOWN);
+    bytes += 1;
+  }
+  return bytes;
+}
+
+}  // namespace
+
+FillNodes::FillNodes(const FillNodesConfig& fill_nodes_config,
+                     const int64 num_operations)
+    : fill_nodes_config_(fill_nodes_config), num_operations_(num_operations) {
+  switch (fill_nodes_config_.specification()) {
+    case FillNodesConfig::ARTIFACT: {
+      name_ = "fill_artifact";
+      break;
+    }
+    case FillNodesConfig::EXECUTION: {
+      name_ = "fill_execution";
+      break;
+    }
+    case FillNodesConfig::CONTEXT: {
+      name_ = "fill_context";
+      break;
+    }
+    default:
+      LOG(FATAL) << "Wrong specification for FillNodes!";
+  }
+  if (fill_nodes_config_.update()) {
+    name_ += "(update)";
+  }
+}
+
+tensorflow::Status FillNodes::SetUpImpl(MetadataStore* store) {
+  LOG(INFO) << "Setting up ...";
+
+  // Gets all the specific types in db to choose from when generating nodes.
+  std::vector<Type> existing_types;
+  TF_RETURN_IF_ERROR(GetAndValidateExistingTypes(
+      fill_nodes_config_.specification(), store, existing_types));
+  std::uniform_int_distribution<int64> uniform_dist_type_index{
+      0, (int64)(existing_types.size() - 1)};
+
+  std::minstd_rand0 gen(absl::ToUnixMillis(absl::Now()));
+
+  std::string node_batch_name;
+  int64 curr_bytes, num_properties, string_value_bytes, num_nodes, type_index;
```

Comment:
move to line 238

struct NodeParam {
    curr_bytes .
    num ...
    string ..
    ..
}

for .. {
    NodeParam param;
}